triton-vm/src/vm.rs

```diff
@@ -136,18 +157,19 @@ pub struct AlgebraicExecutionTrace {
     pub u32_entries: Vec<(Instruction, BFieldElement, BFieldElement)>,
 }
 
-impl Default for AlgebraicExecutionTrace {
-    fn default() -> Self {
+impl AlgebraicExecutionTrace {
+    pub fn new(program: Program) -> Self {
+        let instruction_multiplicities = vec![0_u32; program.len()];
```

Comment:
I know I made `.len()` on `Program`, but in retrospect, this is ambiguous to me:

A program's length could either be the number of instructions, or the number of b-words.

It is actually the number of b-words here, but perhaps `.len()` could be more clear.

Reply:
We could have a method 

```rust
pub fn number_of_instructions(&self) -> usize {
    self.clone().into_iter().count()
}
```

as addition to `.len()`. I decided against that since it would not be used anywhere. I tried to clarify the behavior of `.len()` by introducing the doc-string. Thoughts?

Reply:
We don't have to have a function that never gets called.

I suppose what I'm suggesting is to rename `.len()` into something unambiguous, like `.len_bwords()`.

Reply:
That makes sense and is in line with the naming scheme in `triton-opcodes`. I'll refactor accordingly. :+1:

Reply:
Is there ever a context where we need the number of instructions without counting the arguments?

Reply:
Not that I know of.

Reply:
I guess I'm a little confused about what ambiguity this renaming is supposed to eliminate. What other thing could program length refer to?